Fix the WPT export script

mrobinson opened the pull request:
- Have the WPT exporter script use the WPT_SYNC_TOKEN for checking out
  wpt.
- Make sure the local WPT repository is unshallow when pushing.
- When searching for existing PRs use the main GitHub search API,
  as the pull request search does not seem to properly process
  the "head" parameter.
- When deleting branches in the downstream WPT repository, use
  the full URL to avoid trying to modify the upstream repository.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes OR

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🤖 Opened new upstream WPT pull request (web-platform-tests/wpt#39426) with upstreamable changes.

mrobinson committed:
Fix the WPT export script

- Have the WPT exporter script use the WPT_SYNC_TOKEN for checking out
  wpt.
- Make sure the local WPT repository is unshallow when pushing.
- When searching for existing PRs use the main GitHub search API,
  as the pull request search does not seem to properly process
  the "head" parameter.
- When deleting branches in the downstream WPT repository, use
  the full URL to avoid trying to modify the upstream repository.

delan commented on .github/workflows/upstream-wpt-changes.yml:
```diff
@@ -25,7 +25,11 @@ jobs:
-          token: ${{ secrets.GITHUB_TOKEN }}
+          # The token here must be the token that we will use to push to the
+          # WPT repository and not the token used for GitHub actions, because
+          # the checkout action sets up an extraheader authorization override
```
nit: typo in “extraheader”?

mrobinson replied:
This git setting is actually called `extraheader`. I can add quotes here to make it clearer though and also link to the GitHub actions issue.

delan replied:
yes please!

delan commented on etc/ci/upstream-wpt-changes/wptupstreamer/step.py:
```diff
@@ -198,7 +204,11 @@ def run(self, run: SyncRun):
+            token = run.sync.github_api_token
+            repo = run.sync.downstream_wpt_repo
+            remote_url = f"https://{user}:{token}@github.com/{repo}.git"
+            run.sync.local_wpt_repo.run("push", remote_url, "--delete",
```
will the token in this command be redacted by github actions?

delan replied:
sounds good!

delan reviewed (approved):
Thanks! Looks good, with one question.

mrobinson commented:
@bors-servo r=delan

@delan Thank you for the review!

bors-servo commented:
:pushpin: Commit f4b5b9f has been approved by `delan`

<!-- @bors-servo r=delan f4b5b9f85fe6e4e002a8b935e41f3072f6f19120 -->

mrobinson commented:
@bors-servo retry

bors-servo commented:
:hourglass: Testing commit f4b5b9f85fe6e4e002a8b935e41f3072f6f19120 with merge 15de7b99b7c3d6a54f78f2732ca963a28607a439...
